### DISCUSSION TITLE
Help with first contribution

### DISCUSSION THREAD

**Discussion by ibdul** [withastro/starlight#1679] [Q&A] [answered]
![image](https://github.com/withastro/starlight/assets/26022162/ea9c7152-5110-4aca-a0eb-0e6b2d685c61)

Hi there, I need help running the docs locally. I can't install packages, it gives an error message as depicted in the picture attached. 

Also, how do I preview my changes if made to the main starlight package, I believe that is in /packages/starlight. the scripts only include test scripts. 

Thank you for your help

**Comment by HiDeoo** [comment, marked as answer]
👋 The error seems to be related to the use of `npm` instead of [`pnpm`](https://pnpm.io/) which used by Starlight.

All the prerequisites and steps to develop locally can be found in the [contributing guide](https://github.com/withastro/starlight/blob/main/CONTRIBUTING.md#developing-locally) of the repository. It also includes a [section](https://github.com/withastro/starlight/blob/main/CONTRIBUTING.md#testing-visual-changes-while-you-work) on how to visually test changes while working on the project.

If needed, you can also join the Astro community on [Discord](https://astro.build/chat) in the dedicated `#starlight` channel if you have any more questions or want to discuss potential contributions.

**Reply by ibdul** [reply]
ALright, thank you so much
